File: Medium/2025-11-16-1513-Number-of-Substrings-With-Only-1s/solution.java
class Solution {
    public int numSub(String s) {
        long count = 0;
        long current = 0;
        final int MOD = 1000000007;
        
        for (char c : s.toCharArray()) {
            if (c == '1') {
                current++;
                count = (count + current) % MOD;
            } else {
                current = 0;
            }
        }
        
        return (int) count;
    }
}

File: Medium/2025-11-16-1513-Number-of-Substrings-With-Only-1s/solution.js
var numSub = function(s) {
    const MOD = 1e9 + 7;
    let count = 0, result = 0;

    for (let char of s) {
        if (char === '1') {
            count++;
            result = (result + count) % MOD;
        } else {
            count = 0;
        }
    }

    return result;
};